src/solidity/types/ArrayType.js

```diff
@@ -19,7 +21,42 @@ function ArrayType (underlyingType, arraySize) {
 }
 
 ArrayType.prototype.decodeFromStorage = function (location, storageContent) {
-  return '<not implemented yet>'
+  var ret = []
+  var size = null
+  var slotValue = util.extractValue(location, storageContent, this.storageBytes)
+  if (!slotValue) {
+    return []
+  }
+  var currentLocation = {
+    offset: 0,
+    slot: location.slot
+  }
+  if (this.arraySize === 'dynamic') {
+    size = util.toBN(slotValue)
+    currentLocation.slot = util.dynamicTypePointer(location.slot)
+    if (!storageContent[currentLocation.slot]) {
+      return []
+    }
+  } else {
+    size = new BN(this.arraySize)
+  }
+  var k = util.toBN(0)
+  var one = util.toBN(1)
+  while (k.lt(size)) {
+    if (currentLocation.offset + this.underlyingType.storageBytes > 32) {
+      currentLocation.offset = 0
+      currentLocation.slot = '0x' + util.add(currentLocation.slot, one).toString(16)
```

Comment:
Perhaps keep the slot as a BN so that we do not need to keep converting from and to string all the time but just to string?